crates/proof-impl/btc-blockspace/src/logic.rs

```diff
@@ -2,59 +2,68 @@
 
 use bitcoin::Block;
 use borsh::{BorshDeserialize, BorshSerialize};
-use strata_primitives::{buf::Buf32, params::RollupParams};
+use strata_primitives::{block_credential::CredRule, buf::Buf32};
 use strata_state::{batch::BatchCheckpoint, tx::DepositInfo};
+use strata_tx_parser::filter::TxFilterRule;
 
-use crate::{block::check_merkle_root, filter::extract_relevant_info};
-
-#[derive(Debug)]
-pub struct BlockspaceProofInput {
-    pub block: Block,
-    pub rollup_params: RollupParams,
-    // TODO: add hintings and other necessary params
-}
+use crate::{
+    block::{check_merkle_root, check_witness_commitment},
+    filter::extract_relevant_info,
+};
 
 #[derive(Debug, BorshSerialize, BorshDeserialize)]
 pub struct BlockspaceProofOutput {
     pub header_raw: Vec<u8>,
     pub deposits: Vec<DepositInfo>,
     pub prev_checkpoint: Option<BatchCheckpoint>,
-    pub rollup_params_commitment: Buf32,
+    pub tx_filters_commitment: Buf32,
+    pub cred_rule: CredRule,
```

Comment:
Is this only used for checking checkpoint signatures?  This will have to be refactored when we have multiple sequencers eventually.

Reply:
Yes the `cred_rule` is only used for validating signatures. `tx_filters_commitment` is used so that we don't have to compute it on every blockspace proof. We later verify it is from rollup params in the batch proof.